ambry-api/src/main/java/com/github/ambry/config/MySqlNamedBlobDbConfig.java

```diff
@@ -29,6 +29,47 @@ public class MySqlNamedBlobDbConfig {
   public static final String  QUERY_STALE_DATA_MAX_RESULTS = PREFIX + "query.stale.data.max.results";
   public static final String  STALE_DATA_RETENTION_DAYS = PREFIX + "stale.data.retention.days";
   public static final String TRANSACTION_ISOLATION_LEVEL = PREFIX + "transaction.isolation.level";
+  public static final String LIST_NAMED_BLOBS_SQL = "list.named.blobs.sql";
+
+  /**
+   * List named-blobs query.
+   * The inner query selects blobs based on the filters, and groups them on blob_name.
+   * In each group, it selects the blob with the highest version.
+   * The outer query filters out blobs that have been deleted, and return the latest blob in each group that is ready.
+   */
+  @Config(LIST_NAMED_BLOBS_SQL)
+  public static final String DEFAULT_LIST_NAMED_BLOBS_SQL = ""
+      + " WITH "
+      + "  BlobsAllVersion AS ( "
```

Comment:
I feel like this query is still pretty complex. I suggest we can think of some simpler queries which makes the code easy to read. Sometimes having two queries is also ok cause we are trying to avoid the complex sql queris which might missing edge cases. cc @justinlin-linkedin

Reply:
I assure you this is one of the simplest queries, compared to queries issued elsewhere. It creates two minimal tables and then joins them. The join was unavoidable becuase of our previous design choices. However, we can be careful in join by reducing the number of rows on either side.

Reply:
Yeah, As we discussed, I think your query is clear if we take some time to understand your logic. However, for debugging purpose it would be hard to construct such a long query. But anyway, it can be out of scope of this pr.

Reply:
discussed offline, online query formatters, debugging v realtime performance.